collector/zfs.go

```diff
@@ -0,0 +1,109 @@
+package collector
+
+// +build linux freebsd
+// +build !nozfs
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+type zfsMetricValue int
+
+const zfsErrorValue = zfsMetricValue(-1)
+
+var zfsNotAvailableError = errors.New("ZFS / ZFS statistics are not available")
+
+type zfsSysctl string
+type zfsSubsystemName string
```

Comment:
This is maybe overdoing the special typing a bit, I'd just keep those as normal strings.